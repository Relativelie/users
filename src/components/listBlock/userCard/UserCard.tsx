import { FC } from 'react';
import { Link } from 'react-router-dom';

type Props = {
    id: number,
    name: string,
    address: { city: string },
    company: { name: string },
    openUserInfo: Function
};

export const UserCard: FC<Props> = ({ id, name, address, company, openUserInfo }) => {
    return (
        <div className="list__card">
            <div>
                <p className="list__cardContent">
                    <span className="list__cardContentTitle">ФИО:</span>
                    {name}
                </p>
                <p className="list__cardContent">
                    <span className="list__cardContentTitle">город:</span>
                    {address.city}
                </p>
                <p className="list__cardContent">
                    <span className="list__cardContentTitle">компания:</span>
                    {company.name}
                </p>
            </div>
            <Link
                to={`user/${id}`}
                className="list__more"
                onClick={() => openUserInfo(id)}
            >
                Подробнее
            </Link>
        </div>
    );
};
