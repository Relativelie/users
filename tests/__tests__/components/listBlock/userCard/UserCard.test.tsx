import { screen } from '@testing-library/dom';
import { MemoryRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { UserCard } from '../../../../../src/components/listBlock/userCard/UserCard';
import { render } from '../../../../testsSetup/test-utils';

let openCard: Function;
beforeEach(() => {
    openCard = jest.fn();
});

describe('user card component', () => {
    test('snapshot - user card component', () => {
        const tree = render(
            <Router>
                <UserCard
                    key={1}
                    id={3}
                    name="Alla"
                    company={{ name: 'Lala' }}
                    openCard={openCard}
                    address={{ city: 'Kazan' }}
                />
            </Router>,
        );
        expect(tree).toMatchSnapshot();
    });

    test('click to link More..', () => {
        render(
            <Router>
                <UserCard
                    key={1}
                    id={3}
                    name="Alla"
                    company={{ name: 'Lala' }}
                    openCard={openCard}
                    address={{ city: 'Kazan' }}
                />
            </Router>,
        );
        const editButton = screen.getByText(/подробнее/i);
        userEvent.click(editButton);
        expect(openCard).toHaveBeenCalledTimes(1);
    });
});
