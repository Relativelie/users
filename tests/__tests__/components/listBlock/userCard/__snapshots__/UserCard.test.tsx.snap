// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card block component snapshot - card block component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list__card"
      >
        <div>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              ФИО:
            </span>
            Alla
          </p>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              город:
            </span>
            Kazan
          </p>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              компания:
            </span>
            Lala
          </p>
        </div>
        <a
          class="list__more"
          href="/user/3"
        >
          Подробнее
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list__card"
    >
      <div>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            ФИО:
          </span>
          Alla
        </p>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            город:
          </span>
          Kazan
        </p>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            компания:
          </span>
          Lala
        </p>
      </div>
      <a
        class="list__more"
        href="/user/3"
      >
        Подробнее
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`user card component snapshot - user card component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list__card"
      >
        <div>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              ФИО:
            </span>
            Alla
          </p>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              город:
            </span>
            Kazan
          </p>
          <p
            class="list__cardContent"
          >
            <span
              class="list__cardContentTitle"
            >
              компания:
            </span>
            Lala
          </p>
        </div>
        <a
          class="list__more"
          href="/user/3"
        >
          Подробнее
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list__card"
    >
      <div>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            ФИО:
          </span>
          Alla
        </p>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            город:
          </span>
          Kazan
        </p>
        <p
          class="list__cardContent"
        >
          <span
            class="list__cardContentTitle"
          >
            компания:
          </span>
          Lala
        </p>
      </div>
      <a
        class="list__more"
        href="/user/3"
      >
        Подробнее
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
